lib/OpenQA/Locks.pm

```diff
@@ -32,12 +32,9 @@ sub _get_lock {
     # We need to get owner of the lock
     # owner can be one of the parents or ourselves if we have no parent
     my $lock;
-    my @maybeowners = $job->parents->all || ($job);
-    for (@maybeowners) {
-        $lock = $schema->resultset('JobLocks')->single({name => $name, owner => $_->id});
-        last if ($lock);
-    }
-    return $lock;
+    my @maybeowners;
+    push(@maybeowners, $_->id) for ($job->parents->all, $job);
+    return $schema->resultset('JobLocks')->search({name => $name, owner => { -in => \@maybeowners} })->single;
```

Comment:
Yeah, I had a feeling this is usually solved differently but I'm exceptionally slow today. Will fix.